ldap/servers/plugins/replication/cl5_api.c

```diff
@@ -2589,7 +2589,7 @@ _cl5CICbInit(dbi_val_t *key, dbi_val_t *data, DBLCI_CTX *dblcictx)
         return DBI_RC_SUCCESS;
     }
     /* Update last csn */
-    csn_init_by_string(&dblcictx->csn, data->data);
+    csn_init_by_string(&dblcictx->csn, key->data);
```

Comment:
I understand that fix but I can not understand how it have been working before.
_cl5Iterate is used for Trimming, cleanRuV,.... We know it works. But IIUC it was setting the CL iterator with crappy csn (data instead of key). It looks it worked to access the first CL entry but on next entries,  'startcsn' in _cl5Iterate should have been completely random !
I am missing something here.

Anyway this fix is valid.

Reply:
Yes, it is broken since  5a755180c192c79d65cc13e75ec6c473d7f11883 on Mar  2023 
 Issue 5661 - LMDB hangs while Rebuilding the replication changelog RUV (#5676)
So it is quite recent, 2.4,  I think and this code is only called in case of disorderly shutdown or restore or import
  (i.e: When purge RUV and upperbound Ruv are not found in changelog) 
 so it is not that frequent, and obviously The CI tests were not testing it until now ...

Reply:
Ah !! Okay I thought it was a old (always existing) bug :)
Thanks for the detailed explanation